src/algebra/star/module.lean

```diff
@@ -62,3 +63,27 @@ lemma star_module.eq_self_adjoint_part_add_skew_adjoint_part (x : A) :
   x = self_adjoint_part R x + skew_adjoint_part R x :=
 by simp only [smul_sub, self_adjoint_part_apply_coe, smul_add, skew_adjoint_part_apply_coe,
               add_add_sub_cancel, inv_of_two_smul_add_inv_of_two_smul]
+
+variables (A)
+
+/-- The self-adjoint elements of a star module, as a submodule. -/
+def self_adjoint.submodule : submodule R A :=
+{ smul_mem' := λ r x (hx : star x = x), by simp [self_adjoint.mem_iff, hx],
+  ..self_adjoint A }
+
+/-- The skew-adjoint elements of a star module, as a submodule. -/
+def skew_adjoint.submodule : submodule R A :=
+{ smul_mem' := λ r x (hx : star x = -x), by simp [skew_adjoint.mem_iff, hx],
+  ..skew_adjoint A }
```

Comment:
Instead of using `by simp` here, you can use the `smul_mem` lemma that was recently merged.